scripts/import-wikis-from-local-backup.sh

```diff
@@ -0,0 +1,298 @@
+	fi
+	if [ ! -f "$wiki_install_path/config/favicon.ico" ]; then
+		cp "$m_config/template/wiki-init/config/favicon.ico" "$wiki_install_path/config/favicon.ico"
+	fi
+	if [ ! -f "$wiki_install_path/config/postLocalSettings.php" ]; then
+		# old method used overrides.php...rename that file if it exists
+		if [ -f "$wiki_install_path/config/overrides.php" ]; then
+			mv "$wiki_install_path/config/overrides.php" "$wiki_install_path/config/postLocalSettings.php"
+		else
+			cp "$m_config/template/wiki-init/config/postLocalSettings.php" "$wiki_install_path/config/postLocalSettings.php"
+		fi
+	fi
+	if [ ! -f "$wiki_install_path/config/preLocalSettings.php" ]; then
+		# old method used setup.php...rename that file if it exists
+		if [ -f "$wiki_install_path/config/setup.php" ]; then
+			mv "$wiki_install_path/config/setup.php" "$wiki_install_path/config/preLocalSettings.php"
+		else
+			cp "$m_config/template/wiki-init/config/preLocalSettings.php" "$wiki_install_path/config/preLocalSettings.php"
+		fi
+	fi
+	chmod -R 755 "$wiki_install_path/config"
+
+	# insert wiki name and auth type into preLocalSettings.php if it's still "placeholder"
+	sed -r -i "s/wgSitename = 'placeholder';/wgSitename = '$wiki_name';/g;" "$wiki_install_path/config/preLocalSettings.php"
+
+	# If preLocalSettings.php already existed, it may have a $mezaCustomDBname set.
+	# This import script normalizes all database names to be in the form
+	# "wiki_$wiki_id", so if $wiki_id is "eva" then the database is "wiki_eva"
+	#
+	# This command just comments out the old database name
+	sed -i "s/\$mezaCustomDBname/\/\/ \$mezaCustomDBname/g;" "$wiki_install_path/config/preLocalSettings.php"
+
+	# import SQL file
+	# Import database - Ref: https://www.mediawiki.org/wiki/Manual:Restoring_a_wiki_from_backup
+
+	# determine and use most-recent sql file
+	# Ref: http://stackoverflow.com/a/4447795/5103312
+	import_sql_file="$(find $local_wiki_backup/$wiki_id -maxdepth 1 -type f -iname "*.sql" | sort -r | head -n +1)"
+	wiki_db_name="wiki_$wiki_id"
+	echo "For $wiki_db_name: "
+	echo " * dropping if exists"
+	echo " * (re)creating"
+	echo " * importing file at $import_sql_file"
+	mysql -u root "--password=$mysql_root_pass" -e"DROP DATABASE IF EXISTS $wiki_db_name; CREATE DATABASE $wiki_db_name; use $wiki_db_name; SOURCE $import_sql_file;"
+	#
+	# TO-DO: Remove the following rm command; don't remove sql file from backup dir, just use it
+	# rm -rf "$import_sql_file"
+
+
+	# Run update.php. The database you imported may not be up to the same version
+	# as meza, and thus you must update it.
+	#
+	# TO-DO: update.php commented-out for now; may not be necessary
+	#
+	# echo "Running MediaWiki maintenance script \"update.php\""
+	# WIKI="$wiki_id" php "$m_mediawiki/maintenance/update.php" --quick
+
+
+	# if SMW set up yet. On the very first install it will not be.
+	#
+	# TO-DO: SMW maintenance commented-out for now; may not be necessary
```

Comment:
This whole section is commented out for SMW maint. Again, could have an option to do SMW maint or not.